src/managers/notification-manager.ts

```diff
@@ -21,71 +22,86 @@ export interface ToastActionParams {
 class NotificationManager extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @state() private _action?: ToastActionParams;
+  @state() private _parameters?: ShowToastParams;
 
-  @state() private _noCancelOnOutsideClick = false;
+  @query("ha-toast") private _toast!: HaToast | undefined;
 
-  @query("ha-toast") private _toast!: HaToast;
-
-  public async showDialog({
-    message,
-    action,
-    duration,
-    dismissable,
-  }: ShowToastParams) {
-    let toast = this._toast;
+  public async showDialog(parameters: ShowToastParams) {
     // Can happen on initial load
-    if (!toast) {
+    if (!this._toast) {
       await this.updateComplete;
-      toast = this._toast;
     }
-    toast.setAttribute("dir", computeRTL(this.hass) ? "rtl" : "ltr");
-    this._action = action || undefined;
-    this._noCancelOnOutsideClick =
-      dismissable === undefined ? false : !dismissable;
-    toast.hide();
-    toast.show({
-      text: message,
-      duration: duration === undefined ? 3000 : duration,
-    });
+    this._toast?.close("dismiss");
+    this._parameters = parameters;
+    if (!this._parameters || this._parameters.duration === 0) {
+      this._parameters = undefined;
+      return;
+    }
+
+    if (
+      this._parameters.duration === undefined ||
+      (this._parameters.duration > 0 && this._parameters.duration <= 4000)
+    ) {
+      this._parameters.duration = 4000;
+    }
   }
 
-  protected render(): TemplateResult {
+  public shouldUpdate(changedProperties) {
+    return !this._toast || changedProperties.has("_parameters");
+  }
+
+  public updated() {
+    if (this._parameters) {
+      this._toast?.show();
+    }
+  }
```

Comment:
Is this needed?

Reply:
A toast is not shown by default after it is created. It's needs to be shown. If the toast is being clean up, it can still exist although the parameters are already set undefined. When that happens, it should not show the toast.